docs/40_smart-contract-guides/92_read-only-queries.md

```diff
@@ -0,0 +1,341 @@
+* Access to a running blockchain. Click on this link for instructions on [running a blockchain](01_before-you-begin/10_running-a-blockchain.md)
+* An EOSIO account and access to the account keys. Click on this link for information on [Accounts and Permissions](01_before-you-begin/20_accounts-and-permissions.md)
+* You understand how to build and deploy smart contracts. Follow this link for more information [hello world tutorial.](../30_getting-started-guide/25_hello-world.md)
+
+
+## Run a Local Single-Node Blockchain
+Run a local nodeos instance which produces blocks, creating a local single-node testnet, using these [instructions](https://developers.eos.io/manuals/eosio.cdt/latest/installation/index).
+ 
+
+## Build a Smart Contract with a Read-Only Query 
+
+Create a simple smart contract which contains three read-only actions: 
+- `get` - Will return a value 
+- `getComp` - Will return a complex value
+- `getMore` - Will return a vector of values
+
+Define the actions as read-only with the `eosio::read_only` attribute. Click on the follwing link for more information [ABI code generator attributes explained](https://developers.eos.io/manuals/eosio.cdt/latest/best-practices/abi/abi-code-generator-attributes-explained)   
+
+
+```cpp
+#include <eosio/eosio.hpp>
+class [[eosio::contract]] hello : public eosio::contract {
+  public:
+
+     struct retVal {
+         std::string name;
+         uint32_t gender;
+         uint32_t age;
+      };
+ 
+      using eosio::contract::contract;
+
+
+      [[eosio::action, eosio::read_only]] std::string get() {
+         return "Bob";
+      }
+
+
+      [[eosio::action, eosio::read_only]] retVal getComp() {
+         retVal rv;
+
+         rv.name = "Bob";
+         rv.gender = 1;
+         rv.age = 99;
+
+         return rv;
+      }
+
+
+      [[eosio::action, eosio::read_only]] std::vector<retVal> getMore() {         
+         std::vector<retVal> ret;
+         
+         retVal rv;
+         
+         rv.name = "Bob";
+         rv.gender = 1;
+         rv.age = 99;
+         ret.push_back(rv);
+
+         rv.name = "Alice";
+         rv.gender = 0;
+         rv.age = 19;
+         ret.push_back(rv);
+
+         return ret;
+      }
+};
+```
+
+Build the smart contract and deploy the smart contract to the local single-node testnet using [how to compile a smart contract via the CLI](https://developers.eos.io/manuals/eosio.cdt/latest/how-to-guides/compile/compile-a-contract-via-cli) and [how to deploy a smart contract.](https://developers.eos.io/manuals/eos/latest/cleos/how-to-guides/how-to-deploy-a-smart-contract) 
+
+## Call the Read-Only Query 
+Once you have deployed the smart contract you can test it by calling the actions and seeing the data returned. Assuming the smart contract is deployed to an account called _scholder_ and that you have an account named _bob_ to authorise calling the actions. 
```

Comment:
I appreciate the callout, but it not be a sentence fragment?

Reply:
not sure what you mean here ...

Reply:
"Assuming the smart contract is deployed to an account called _scholder_ and that you have an account named _bob_ to authorise calling the actions. "

That is not a sentence.  It is a fragment.